[PyTorch][Edge] Tests for QuantizationFx API on lite modules

pavithranrao opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#60476**

# Context
Add tests for Lite modules that are quantized using fx API

Differential Revision: [D29306402](https://our.internmc.facebook.com/intern/diff/D29306402/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D29306402/)!

Commit:
[PyTorch][Edge] Tests for QuantizationFx API on lite modules

# Context
Add tests for Lite modules that are quantized using fx API

Differential Revision: [D29306402](https://our.internmc.facebook.com/intern/diff/D29306402/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D29306402/)!

[ghstack-poisoned]

Commit:
Update on "[PyTorch][Edge] Tests for QuantizationFx API on lite modules"


# Context
Add tests for Lite modules that are quantized using fx API

Differential Revision: [D29306402](https://our.internmc.facebook.com/intern/diff/D29306402/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D29306402/)!

[ghstack-poisoned]

Commit:
Update on "[PyTorch][Edge] Tests for QuantizationFx API on lite modules"


# Context
Add tests for Lite modules that are quantized using fx API

Differential Revision: [D29306402](https://our.internmc.facebook.com/intern/diff/D29306402/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D29306402/)!

[ghstack-poisoned]

Commit:
Update base for Update on "[PyTorch][Edge] Tests for QuantizationFx API on lite modules"


# Context
Add tests for Lite modules that are quantized using fx API

Differential Revision: [D29306402](https://our.internmc.facebook.com/intern/diff/D29306402/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D29306402/)!

[ghstack-poisoned]

Commit:
Update on "[PyTorch][Edge] Tests for QuantizationFx API on lite modules"


# Context
Add tests for Lite modules that are quantized using fx API

Differential Revision: [D29306402](https://our.internmc.facebook.com/intern/diff/D29306402/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D29306402/)!

[ghstack-poisoned]

iseeyuan commented on test/mobile/test_quantize_fx_lite_script_module.py:
```diff
@@ -0,0 +1,77 @@
+        data = torch.randn(1, 1, 1, 1)
+        m = convert_fx(m)
+        # first conv is quantized, second conv is not quantized
+        node_list = [
```
Looks like it's not used?

Commit:
Update base for Update on "[PyTorch][Edge] Tests for QuantizationFx API on lite modules"


# Context
Add tests for Lite modules that are quantized using fx API

Differential Revision: [D29306402](https://our.internmc.facebook.com/intern/diff/D29306402/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D29306402/)!

[ghstack-poisoned]

Commit:
Update on "[PyTorch][Edge] Tests for QuantizationFx API on lite modules"


# Context
Add tests for Lite modules that are quantized using fx API

Differential Revision: [D29306402](https://our.internmc.facebook.com/intern/diff/D29306402/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D29306402/)!

[ghstack-poisoned]

Commit:
Update base for Update on "[PyTorch][Edge] Tests for QuantizationFx API on lite modules"


# Context
Add tests for Lite modules that are quantized using fx API

Differential Revision: [D29306402](https://our.internmc.facebook.com/intern/diff/D29306402/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D29306402/)!

[ghstack-poisoned]

Commit:
Update on "[PyTorch][Edge] Tests for QuantizationFx API on lite modules"


# Context
Add tests for Lite modules that are quantized using fx API

Differential Revision: [D29306402](https://our.internmc.facebook.com/intern/diff/D29306402/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D29306402/)!

[ghstack-poisoned]

iseeyuan commented on test/mobile/test_quantize_fx_lite_script_module.py:
```diff
@@ -0,0 +1,95 @@
+                return x
+
+        m = M().eval()
+        qconfig_dict = {"": default_qconfig, "module_name": [("conv1", None)]}
```
Thanks for this update!

iseeyuan reviewed (approved):
LGTM! Would be great if @jerryzh168 and @supriyar could take a second look.

supriyar commented on test/mobile/test_quantize_fx_lite_script_module.py:
```diff
@@ -0,0 +1,93 @@
+)
+
+
+class TestFuseFx(QuantizationLiteTestCase):
```
Why is this class called TestFuseFx?

supriyar reviewed (approved):
looks good!

Commit:
Update base for Update on "[PyTorch][Edge] Tests for QuantizationFx API on lite modules"


# Context
Add tests for Lite modules that are quantized using fx API

Differential Revision: [D29306402](https://our.internmc.facebook.com/intern/diff/D29306402/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D29306402/)!

[ghstack-poisoned]

Commit:
Update on "[PyTorch][Edge] Tests for QuantizationFx API on lite modules"


# Context
Add tests for Lite modules that are quantized using fx API

Differential Revision: [D29306402](https://our.internmc.facebook.com/intern/diff/D29306402/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D29306402/)!

[ghstack-poisoned]

Commit:
Update base for Update on "[PyTorch][Edge] Tests for QuantizationFx API on lite modules"


# Context
Add tests for Lite modules that are quantized using fx API

Differential Revision: [D29306402](https://our.internmc.facebook.com/intern/diff/D29306402/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D29306402/)!

[ghstack-poisoned]

Commit:
Update on "[PyTorch][Edge] Tests for QuantizationFx API on lite modules"


# Context
Add tests for Lite modules that are quantized using fx API

Differential Revision: [D29306402](https://our.internmc.facebook.com/intern/diff/D29306402/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D29306402/)!

[ghstack-poisoned]
